### DISCUSSION TITLE
Is it possible to use more than one JEELIZFACEFILTER instance?

### DISCUSSION THREAD

**Discussion by jmakar-boom** [jeeliz/jeelizFaceFilter#270] [Q&A] [answered]
I am interested in using JEELIZFACEFILTER on more than one video simultaneously. It appears to be a singleton though. I am aware of the MediaStream demo which swaps the current video for another. But for the use case of multiple simultaneous videos, is there a path forward for that with the current version of the library?

Thanks for any input.

**Comment by xavierjs** [comment, marked as answer]
Hi yes it is possible please check: https://github.com/jeeliz/jeelizFaceFilter#multiple-videos
However, be careful about the computing power. If you run facefilter for N videos it will require N times computing power (GPU power, the GPU is the limiting resource most often). Video resolution should be as low as possible.

If it is for a video call application, where participants should have face filters over the face, it is really better that each user run Jeeliz FaceFilter only on its own video, then you stream the video with the facefilter on than if each user run Jeeliz Facefilter for all the videos of the other users. It is better for performance, but also for privacy (it is not possible to remove the masks).

**Reply by jmakar-boom** [reply]
Thanks for the prompt response. I didn't realize you could stream the video with the filter already applied. I must have missed that (if the MediaStream demo illustrates it). That would be the best approach. Great job with these libs!

I'll also check out the link you supplied.
